galleries/examples/images_contours_and_fields/watermark_image.py

```diff
@@ -17,9 +18,16 @@
 
 fig, ax = plt.subplots()
 
-ax.plot(np.sin(10 * np.linspace(0, 1)), '-o', ms=20, alpha=0.7, mfc='orange')
+np.random.seed(19680801)
+x = np.arange(30)
+y = np.sin(x) + x + np.random.randn(30)
+ax.bar(x, y, color='#6bbc6b')
 ax.grid()
-fig.figimage(im, 10, 10, zorder=3, alpha=.5)
+
+image_size = (542, 130)
+nx, ny = fig.get_size_inches() * fig.dpi
+
+fig.figimage(im, (nx - image_size[0]) / 2, (ny - image_size[1]) / 2, zorder=3, alpha=.7)
```

Comment:
I think figimage is buggy, so probably shouldn't be used at all?  https://github.com/matplotlib/matplotlib/issues/23399

On my hi-dpi screen this is half size:

<img width="637" alt="Watermark" src="https://user-images.githubusercontent.com/1562854/232322684-78a8abad-0e2d-4adc-97ae-96fe9ac21dea.png">

```suggestion
axin = fig.add_axes([0, 0, 1, 1])
axin.axis('off')
axin.imshow(im, alpha=0.7)
```

Reply:
I'm a bit torn here. Overlaying an Axes and plotting the image into that feels like a hacksish workaround workaround. OTOH  `figimage` has currently it's deficits. (Not only the dpi issue, but also API wise: One would want a proper positioning logic (horizontal/vertical align, bbox_to_anchor, sizing).

What I additionally don't like about the Axes/imshow approach is that you get sampling artifacts unless you make sure the image is exactly the same size as the figure.

I will rethink how to approach this.

Reply:
> What I additionally don't like about the Axes/imshow approach is that you get sampling artifacts unless you make sure the image is exactly the same size as the figure.

I'm not sure I understand what you are referring to here.  

I don't personally consider placing Artists in an Axes rather than on a Figure a "hack".  Almost all our artists are created on Axes, and I don't think that there is much to be gained by making figure-equivalent versions of most of those methods.  

I _do_ consider `axis.axis` to be a bit clunky, but so far as I know that is the one-line way to make the axes disappear.  

If we were to add an API, it might be `fig.figure_axes` which just does `ax = fig.add_axes([0, 0, 1, 1]); ax.axis('off')` to give a figure-wide canvas.

Reply:
> > What I additionally don't like about the Axes/imshow approach is that you get sampling artifacts unless you make sure the image is exactly the same size as the figure.
> 
> I'm not sure I understand what you are referring to here.

`figimage` only gets the anchor point an then draws the image 1:1 on the pixels -> every image pixel appears 1:1 in the figure. In constrast, for imshow the image is scaled to the size of the Axes. This scaling creates artifacts. While scaling can be a nice feature, there should be a way to have images rendered 1:1 on the pixels.

Reply:
This is a watermark example - I would have thought we would want the watermark to rescale if the figure changes size?

Figimage indeed does what it says, which is "Add a non-resampled image to the figure.", but the example doesn't properly query the dpi or create a figure that is the right size for the logo png.  As a general sniff test, if an example needs `fig.dpi` it's likely to be incorrect responding to figure size changes or screen resolutions.